### PR TITLE
recastnavigation: add v1.6.0, simplify patching

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

**Comment by RubenRBS** [comment]
Closing this one, please read https://github.com/conan-io/conan-center-index/pull/20321#issuecomment-1744976935 !

File: recipes/recastnavigation/all/patches/001_fix_shared_option.patch [hunk 1]
```diff
@@ -1,129 +0,0 @@
...
--endif()
-+add_library(DebugUtils ${SOURCES})
-
- add_library(RecastNavigation::DebugUtils ALIAS DebugUtils)
-
-@@ -26,6 +22,7 @@
-         )
-
- install(TARGETS DebugUtils
-+		RUNTIME DESTINATION bin
```
Comment (SpaceIm) [inline recipes/recastnavigation/all/patches/001_fix_shared_option.patch:38]: I think you can't remove this part of the patch.

File: recipes/recastnavigation/all/conanfile.py [hunk 2]
```diff
@@ -26,36 +26,37 @@ class RecastNavigationConan(ConanFile):
...
+        tc.cache_variables["RECASTNAVIGATION_EXAMPLES"] = False
+        tc.cache_variables["RECASTNAVIGATION_STATIC"] = not self.options.shared
+        tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = self.options.shared
         tc.generate()
 
+    def _patch_sources(self):
+        if self.version == "cci.20200511":
+            replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                            "project(RecastNavigation)",
+                            "project(RecastNavigation)\ninclude(GNUInstallDirs)")
```
Comment (SpaceIm) [inline recipes/recastnavigation/all/conanfile.py:56]: Would be better placed in a patch to keep recipe code as simple as possible.

Comment (valgur) [reply recipes/recastnavigation/all/conanfile.py:56]: I agree, I'll move it to a patch after the build is finished to see whether `include(GNUInstallDirs)` is sufficient for correct DLL output (it should be).

Comment (SpaceIm) [reply recipes/recastnavigation/all/conanfile.py:56]: It won't be sufficient

**Review by RubenRBS — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**